### ISSUE TITLE
Posix Parameter Expansion - Plus Form is Triggering SC2154

### ISSUE THREAD

**Issue by kins-dev** [koalaman/shellcheck#2172]
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

```bash
#!/usr/bin/env bash
echo "${debug1-word}"
echo "${debug2=word}"
echo "${debug3?word}"
echo "${debug4+word}"
echo "${debug5:-word}"
echo "${debug6:=word}"
echo "${debug7:?word}"
echo "${debug8:+word}"
```

#### Here's what shellcheck currently says:
```plain
$ shellcheck myscript
 
Line 5:
echo "${debug4+word}"
      ^-- SC2154: debug4 is referenced but not assigned (did you mean 'debug2'?).
 
Line 9:
echo "${debug8:+word}"
      ^-- SC2154: debug8 is referenced but not assigned (did you mean 'debug2'?).

```

#### Here's what I wanted or expected to see:
```plain
$ shellcheck myscript
No issues detected!

```
#### Rationale:
The other forms of [Posix Parameter Expansion](https://pubs.opengroup.org/onlinepubs/009604499/utilities/xcu_chap02.html#tag_02_06_02) are covered but the + form is not.

This form is very useful in scripts where you want to optionally add debugging information if a debug variable is set, for example:
```bash
#!/usr/bin/env bash
set -ue${debug+xv}
```

While one can argue using all uppercase is a reasonable workaround, it fails if `--enable=all` is set.
